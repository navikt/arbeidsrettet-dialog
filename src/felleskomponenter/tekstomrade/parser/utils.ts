import { AST, ASTNode } from './domain';

export type RegexMatch = {
    fullmatch: string;
    capture: Array<string>;
    index: number;
    input: string;
};

export function match(regex: RegExp, content: string): null | RegexMatch {
    const res = regex.exec(content);
    if (!res) {
        return null;
    }
    const [fullmatch, ...capture] = res;
    return {
        fullmatch,
        capture,
        index: res.index,
        input: res.input
    };
}

export function minBy<T>(fn: (t: T) => number) {
    return (acc: null | T, value: T): null | T => {
        if (acc === null) {
            return value;
        }
        const accFn = fn(acc);
        const valueFn = fn(value);

        return accFn < valueFn ? acc : value;
    };
}

export function getText(node: AST | ASTNode): string {
    if (typeof node === 'string') {
        return node;
    } else if (Array.isArray(node)) {
        return node.map(getText).join(' ');
    } else {
        return node.content.map(getText).join(' ');
    }
}
