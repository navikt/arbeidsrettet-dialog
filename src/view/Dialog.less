@import (reference) '~nav-frontend-core/less/core';

.dialog {
    flex-grow: 1;
    background: @navLysGra;
    display: flex;
    flex-direction: column;
    margin: 0.5rem;

    .checkbox-block {
        display: flex;
        flex-direction: row;
        margin-top: 0.5rem;
        border-style: solid;
        border-width: 0 0 0.2rem 0;
        &__item {
            flex-grow: 1;
        }
    }
    &__aktivitetskortpreview {
        background-color: @navBlaLighten60;
        height: 4rem;
    }
    @media (min-width: 1200px) {
        &__aktivitetskortpreview {
            display: none;
        }
    }
}

.skriv-melding {
    flex-grow: 1;
    font-weight: bold;
    border-style: solid;
    border-width: 0 0 0.2rem 0;
}

.meldingsfelt {
    width: 100%;
}
