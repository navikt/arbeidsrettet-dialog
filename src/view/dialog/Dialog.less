@import (reference) '~nav-frontend-core/less/core';

.dialog {
    display: flex;
    flex-direction: column;
    height: 100%;
    width: 100%;
    background: @white;

    &__header {
        background-color: @navLysGra;
        flex-basis: 4rem;
        box-shadow: 0 5px 15px rgba(0, 0, 0, 0.2);
        z-index: 100;
        .dialog__tittel {
            padding: 1rem;
        }
    }

    .tilbake-til-oversikt {
        text-decoration: none;
        padding: 1rem 0.5rem 0.2rem 0.5rem;
        display: flex;
        align-items: center;
        color: @navMorkGra;
        font-weight: bold;
    }

    @media (min-width: 800px) {
        .tilbake-til-oversikt {
            display: none;
        }
        &__header {
            box-shadow: none;
        }
        &__tom {
            display: block;
            margin: 0 auto;
            text-align: center;
            padding-top: 8rem;
            width: 19rem;
            .infotekst {
                text-align: center;
                margin-top: 1rem;
            }
            svg {
                width: 10rem;
            }
        }
    }
    @media (min-width: 1200px) {
        &__header {
            display: none;
        }
    }
}

.henvendelse-list {
    padding: 1rem;
    flex: 1;
    overflow-y: scroll;
    border-bottom: 2px solid @navLysGra;

    &__viewport {
        height: 100%;
        display: flex;
        flex-direction: column;
    }

    &__henvendelse:first-child {
        margin-top: auto;
    }
}

.skriv-melding {
    display: flex;
    flex-basis: 7rem;
    bottom: 0;
    padding: 1rem;
    align-items: flex-end;
    .autosizing-textarea {
        flex: 1;
        max-height: 7rem;
        @media (min-width: 800px) {
            max-height: 15rem;
        }
    }
    .skjemaelement {
        flex: 1;
        margin-bottom: 0;
    }
    .textarea__container label {
        .sr-only();
    }
    .knapp--hoved {
        padding: 0.5rem;
        height: 2.6rem;
        margin-left: 1rem;
    }
}

.meldingsfelt {
    width: 100%;
}
.checkbox-block {
    display: flex;
    padding-left: 1rem;
    margin-top: 1rem;
    &__item {
        padding-right: 2rem;
    }
}
