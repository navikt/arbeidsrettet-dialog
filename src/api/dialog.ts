import { fetchData, fnrQuery } from '../utils/Fetch';
import { DialogData } from '../utils/Typer';
import { baseApiPath } from '../utils/UseApiBasePath';

export function nyDialog(fnr: string | undefined, melding: string, tema: string, aktivitetId?: string) {
    return sendMelding(fnr, melding, tema, undefined, aktivitetId);
}

export function nyHenvendelse(fnr: string | undefined, melding: string, dialogId: string) {
    return sendMelding(fnr, melding, undefined, dialogId);
}

function sendMelding(fnr: string | undefined, melding: string, tema?: string, dialogId?: string, aktivitetId?: string) {
    const query = fnrQuery(fnr);
    const apiBasePath = baseApiPath(fnr);

    const nyDialogData = {
        dialogId: dialogId,
        overskrift: tema,
        tekst: melding,
        aktivitetId: aktivitetId
    };

    return fetchData<DialogData>(`${apiBasePath}/veilarbdialog/api/dialog${query}`, {
        method: 'post',
        body: JSON.stringify(nyDialogData)
    });
}

export function oppdaterFerdigBehandlet(fnr: string | undefined, dialogId: string, ferdigBehandlet: boolean) {
    const query = fnrQuery(fnr);
    const apiBasePath = baseApiPath(fnr);

    return fetchData<DialogData>(
        `${apiBasePath}/veilarbdialog/api/dialog/${dialogId}/ferdigbehandlet/${ferdigBehandlet}${query}`,
        {
            method: 'put'
        }
    );
}

export function oppdaterVenterPaSvar(fnr: string | undefined, dialogId: string, venterPaSvar: boolean) {
    const query = fnrQuery(fnr);
    const apiBasePath = baseApiPath(fnr);

    return fetchData(`${apiBasePath}/veilarbdialog/api/dialog/${dialogId}/venter_pa_svar/${venterPaSvar}${query}`, {
        method: 'put'
    });
}
