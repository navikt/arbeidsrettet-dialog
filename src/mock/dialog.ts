import { DialogData, HenvendelseData, NyDialogMeldingData } from '../utils/typer';
import { rndId } from './utils';
import { JSONArray, JSONObject, ResponseData } from 'yet-another-fetch-mock';

const dialoger: DialogData[] & JSONArray = [
    {
        id: '1',
        overskrift: 'Memes',
        sisteTekst: 'Hei. Hva er status her? Har du finnet Kaptain Sabeltann?',
        sisteDato: '2018-01-28T12:48:56.097+01:00',
        opprettetDato: '2018-02-27T12:48:56.081+01:00',
        historisk: false,
        lest: true,
        venterPaSvar: false,
        ferdigBehandlet: false,
        lestAvBrukerTidspunkt: null,
        erLestAvBruker: false,
        aktivitetId: null,
        henvendelser: [
            {
                id: '1',
                dialogId: '1',
                avsender: 'VEILEDER',
                avsenderId: 'Z123456',
                sendt: '2018-02-27T12:48:56.097+01:00',
                lest: true,
                tekst: 'Hei. Hva er status her? Har du finnet Kaptain Sabeltann?'
            },
            {
                id: '2',
                dialogId: '1',
                avsender: 'BRUKER',
                avsenderId: '0102030405',
                sendt: '2018-02-28T12:48:56.097+01:00',
                lest: true,
                tekst: 'Hei. Leter enda på sjøen :)'
            }
        ],
        egenskaper: []
    },
    {
        id: '3',
        overskrift: 'NOT USED',
        sisteTekst:
            'Det er viktig at du gjennomfører denne aktiviteten med NAV. Gjør du ikke det, kan det medføre at stønaden du mottar fra NAV bortfaller for en periode eller stanses. Hvis du ikke kan gjennomføre aktiviteten, ber vi deg ta kontakt med veilederen din så snart som mulig.',
        sisteDato: '2018-11-21T13:13:20.685+01:00',
        opprettetDato: '2018-11-21T13:13:20.663+01:00',
        historisk: false,
        lest: false,
        venterPaSvar: false,
        ferdigBehandlet: true,
        lestAvBrukerTidspunkt: null,
        erLestAvBruker: false,
        aktivitetId: null,
        henvendelser: [
            {
                id: '4',
                dialogId: '3',
                avsender: 'VEILEDER',
                avsenderId: 'Z990286',
                sendt: '2018-11-21T13:13:20.685+01:00',
                lest: true,
                tekst:
                    'Det er viktig at du gjennomfører denne aktiviteten med NAV. Gjør du ikke det, kan det medføre at stønaden du mottar fra NAV bortfaller for en periode eller stanses. Hvis du ikke kan gjennomføre aktiviteten, ber vi deg ta kontakt med veilederen din så snart som mulig.'
            }
        ],
        egenskaper: ['PARAGRAF8']
    },
    {
        id: '2',
        overskrift: 'Du har fått et varsel fra NAV',
        sisteTekst: 'Jeg har ikke hørt noe fra deg i det siste. Har du forlist?\n',
        sisteDato: '2018-02-01T11:52:20.615+01:00',
        opprettetDato: '2018-02-01T11:52:20.535+01:00',
        historisk: false,
        lest: true,
        venterPaSvar: true,
        ferdigBehandlet: true,
        lestAvBrukerTidspunkt: null,
        erLestAvBruker: false,
        aktivitetId: null,
        henvendelser: [
            {
                id: '3',
                dialogId: '2',
                avsender: 'VEILEDER',
                avsenderId: 'Z123456',
                sendt: '2018-02-01T11:52:20.615+01:00',
                lest: true,
                tekst: 'Jeg har ikke hørt noe fra deg i det siste. Har du forlist?\n'
            }
        ],
        egenskaper: ['ESKALERINGSVARSEL']
    },
    {
        id: '4',
        overskrift: 'Automatiske dialoger',
        sisteTekst:
            'Hei!\n' +
            'Du er registrert som arbeidssøker og NAV trenger å bli kjent med ditt behov for hjelp fra oss, slik at vi kan gi deg riktig veiledning.\n' +
            'Hva mener du? Klik her og vurder hva du selv tenker https://behovsvurdering.nav.no\n',
        sisteDato: '2018-01-28T12:48:56.097+01:00',
        opprettetDato: '2018-02-27T12:48:56.081+01:00',
        historisk: false,
        lest: false,
        venterPaSvar: false,
        ferdigBehandlet: true,
        lestAvBrukerTidspunkt: null,
        erLestAvBruker: false,
        aktivitetId: '1',
        henvendelser: [
            {
                id: '4',
                dialogId: '4',
                avsender: 'VEILEDER',
                avsenderId: 'Z999999',
                sendt: '2018-02-27T12:48:56.097+01:00',
                lest: false,
                tekst:
                    'Hei!\n' +
                    'Du er registrert som arbeidssøker og NAV trenger å bli kjent med ditt behov for hjelp fra oss, slik at vi kan gi deg riktig veiledning.\n' +
                    'Hva mener du? Klik her og vurder hva du selv tenker https://behovsvurdering.nav.no\n'
            },
            {
                id: '5',
                dialogId: '4',
                avsender: 'VEILEDER',
                avsenderId: 'Z999999',
                sendt: '2018-02-28T12:48:56.097+01:00',
                lest: false,
                tekst:
                    'Hei!\n' +
                    'Du har svart at du har utfordringer som hindrer deg i å søke eller være i jobb. Vi vil veilede deg videre og trenger derfor å vite litt mer.\n' +
                    'Du kan velge om du vil fortelle om situasjonen din \n' +
                    '- i et møte med veilederen din på NAV-kontoret\n' +
                    '- i en telefonsamtale\n' +
                    '- her i dialogen\n' +
                    'Skriv svaret ditt i feltet under. Hvis du velger "her i dialogen", kan du fortelle mer allerede nå.\n'
            },
            {
                id: '6',
                dialogId: '4',
                avsender: 'VEILEDER',
                avsenderId: 'Z999999',
                sendt: '2018-02-28T12:48:56.097+01:00',
                lest: false,
                tekst:
                    'Hei!\n' +
                    'Du har svart at du trenger mer veiledning nå som retten til sykepenger nærmer seg slutten. Vi vil veilede deg videre og trenger derfor å vite litt mer.\n' +
                    'Du kan velge om du vil fortelle om situasjonen din \n' +
                    '- i et møte med veilederen din på NAV-kontoret\n' +
                    '- i en telefonsamtale\n' +
                    '- her i dialogen\n' +
                    'Skriv svaret ditt i feltet under. Hvis du velger "her i dialogen", kan du fortelle mer allerede nå.\n'
            }
        ],
        egenskaper: []
    }
];

export function lesDialog(dialogId: string): Promise<ResponseData> {
    const dialog = dialoger.find(dialog => dialog.id === dialogId);
    if (dialog) {
        dialog.lest = true;
        return Promise.resolve({ status: 200 });
    }
    return Promise.resolve({ status: 404 });
}

export function opprettEllerOppdaterDialog(update: NyDialogMeldingData): DialogData & JSONObject {
    const dialogId = !update.dialogId || update.dialogId === '' ? rndId() : `${update.dialogId}`;
    const nyHenvendelse: HenvendelseData = {
        id: rndId(),
        dialogId: dialogId,
        avsender: 'BRUKER',
        avsenderId: 'Z123456',
        sendt: new Date().toISOString(),
        lest: true,
        tekst: update.tekst
    };

    const eksisterendeDialog = dialoger.find(dialog => update.dialogId !== undefined && dialog.id === dialogId);

    if (eksisterendeDialog) {
        const oldDialog = eksisterendeDialog;
        oldDialog.sisteTekst = update.tekst;
        oldDialog.sisteDato = nyHenvendelse.sendt;
        oldDialog.henvendelser.push(nyHenvendelse);
        return oldDialog as DialogData & JSONObject;
    } else {
        const nyDialog: DialogData = {
            id: nyHenvendelse.dialogId,
            overskrift:
                update.overskrift === undefined || update.overskrift === null ? rndId().toString() : update.overskrift,
            sisteTekst: update.tekst,
            sisteDato: new Date().toISOString(),
            opprettetDato: new Date().toISOString(),
            historisk: false,
            lest: true,
            venterPaSvar: true,
            ferdigBehandlet: false,
            lestAvBrukerTidspunkt: null,
            erLestAvBruker: false,
            aktivitetId: null,
            henvendelser: [nyHenvendelse],
            egenskaper: []
        };
        dialoger.push(nyDialog);
        return nyDialog as DialogData & JSONObject;
    }
}

export default dialoger;
