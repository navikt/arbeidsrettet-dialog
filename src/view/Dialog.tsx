import React, { useEffect } from 'react';
import { HenvendelseList } from './HenvendelseList';
import { DialogHeader } from './DialogHeader';
import { useDialogContext, useOppfolgingContext } from '../Context';
import { RouteComponentProps, withRouter } from 'react-router';
import { Aktivitetskort } from './Aktivitetskort';
import { Innholdstittel, Normaltekst } from 'nav-frontend-typografi';
import './Dialog.less';
import DialogInputBoxVisible from "./DialogInputBox";

interface Props extends RouteComponentProps<{ dialogId?: string }> {}

export function Dialog(props: Props) {
    const oppfolgingData = useOppfolgingContext();
    const dialoger = useDialogContext();
    const dialogId = props.match.params.dialogId;
    const valgtDialog = dialoger.data!.find(dialog => dialog.id === dialogId);

    useEffect(() => {
        if (valgtDialog && !valgtDialog.lest) {
            fetch('/veilarbdialog/api/dialog/lest', {
                method: 'PUT',
                body: JSON.stringify({
                    lest: true,
                    dialogId: valgtDialog.id
                })
            }).then(() => dialoger.refetch());
        }
        // eslint-disable-next-line react-hooks/exhaustive-deps
    }, [valgtDialog]);

    if (dialoger.data && dialoger.data.length === 0) {
        return (
            <div className="dialog">
                <Innholdstittel>Dialog</Innholdstittel>
                <Normaltekst>Her kan du sende melding til veilederen din om arbeid og oppfølging.</Normaltekst>
                <Normaltekst>Du kan forvente svar i løpet av noen dager.</Normaltekst>
                <Normaltekst>Klikk på "Ny dialog"</Normaltekst>
            </div>
        );
    } else {
        if (!valgtDialog) {
            return (
                <div className="dialog">
                    <Innholdstittel>Dialog</Innholdstittel>
                    <Normaltekst>Detaljer for valgt dialog vises her.</Normaltekst>
                    <Normaltekst> - Du kan velge en dialog fra oversikten, </Normaltekst>
                    <Normaltekst> - Du kan klikke på "Ny Dialog" for å starte en ny en.</Normaltekst>
                </div>
            );
        }
        return (
            <>
                <div className="dialog">
                    <DialogHeader dialog={valgtDialog} />
                    <HenvendelseList henvendelseDataList={valgtDialog.henvendelser} />
                    <DialogInputBoxVisible dialog={valgtDialog} visible={oppfolgingData!.underOppfolging} />
                </div>
                <Aktivitetskort dialog={valgtDialog} />
            </>
        );
    }
}

export default withRouter(Dialog);