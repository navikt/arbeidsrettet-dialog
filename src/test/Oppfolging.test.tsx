import React from 'react';
import { mount } from 'enzyme';
import { MemoryRouter } from 'react-router';
import * as AppContext from '../view/Provider';
import * as DialogProvider from '../view/DialogProvider';
import { DialogDataProviderType } from '../view/DialogProvider';
import { Dialog } from '../view/dialog/Dialog';
import { Bruker, DialogData, OppfolgingData, PeriodeData } from '../utils/Typer';
import { HenvendelseList } from '../view/henvendelse/HenvendelseList';
import { DialogInputBox } from '../view/dialog/DialogInputBox';
import { DialogHeader } from '../view/dialog/DialogHeader';
import DialogListe from '../view/dialogliste/DialogListe';
import { NyDialogLink } from '../view/dialogliste/NyDialogLink';
import DialogPreview from '../view/dialogliste/DialogPreview';
import { Checkbox } from 'nav-frontend-skjema';
import { FetchResult, Status as FetchMockStatus } from '@nutgaard/use-fetch';
import '../utils/SetupEnzyme';
import DialogContainer from '../view/dialog/DialogContainer';
import DialogOversikt from '../view/dialogliste/DialogOversikt';

const userInfo: Bruker = { id: '010101', erVeileder: true, erBruker: false };
const oppfPerioder: PeriodeData[] = [];
const oppfolgingData: OppfolgingData = {
    fnr: 'null',
    veilederId: '101010',
    reservasjonKRR: false,
    manuell: false,
    underOppfolging: true, //eller false
    underKvp: false,
    oppfolgingUtgang: null,
    gjeldendeEskaleringsvarsel: null,
    kanStarteOppfolging: false,
    avslutningStatus: null,
    oppfolgingsPerioder: oppfPerioder,
    harSkriveTilgang: true,
    kanReaktiveres: false,
    inaktiveringsdato: '2018-08-31T10:46:10.971+01:00',
    aktorId: 'null',
    erSykmeldtMedArbeidsgiver: false,
    formidlingsgruppe: null,
    kanVarsles: true,
    servicegruppe: null
};
const useFetchOppfolging: FetchResult<OppfolgingData> = {
    status: FetchMockStatus.OK,
    statusCode: 0,
    data: oppfolgingData,
    rerun(): void {}
};
const dialoger = [
    {
        id: '1',
        aktivitetId: '1',
        overskrift: 'Memes',
        sisteTekst: 'Hei. Hva er status her? Har du finnet Kaptain Sabeltann?',
        sisteDato: '2018-01-28T12:48:56.097+01:00',
        opprettetDato: '2018-02-27T12:48:56.081+01:00',
        historisk: false,
        lest: true,
        venterPaSvar: false,
        ferdigBehandlet: false,
        lestAvBrukerTidspunkt: null,
        erLestAvBruker: false,
        henvendelser: [
            {
                id: '1',
                dialogId: '1',
                avsender: 'VEILEDER',
                avsenderId: 'Z123456',
                sendt: '2018-02-27T12:48:56.097+01:00',
                lest: true,
                tekst: 'Hei. Hva er status her? Har du finnet Kaptain Sabeltann?'
            },
            {
                id: '2',
                dialogId: '1',
                avsender: 'BRUKER',
                avsenderId: '0102030405',
                sendt: '2018-02-28T12:48:56.097+01:00',
                lest: true,
                tekst: 'Hei. Leter enda på sjøen :)'
            }
        ],
        egenskaper: []
    }
];
const useDialogContext: DialogDataProviderType = {
    status: 3,
    dialoger: dialoger,
    hentDialoger: () => Promise.resolve([]),
    nyDialog: (melding: string, tema: string, aktivitetId?: string) => Promise.resolve({} as any),
    nyHenvendelse: (melding: string, dialog: DialogData) => Promise.resolve(dialog),
    lesDialog: (dialog: DialogData) => Promise.resolve(dialog),
    setFerdigBehandlet: (dialog: DialogData, ferdigBehandlet: boolean) => Promise.resolve(dialog),
    setVenterPaSvar: (dialog: DialogData, venterPaSvar: boolean) => Promise.resolve(dialog)
};

describe('<DialogContainer/>', () => {
    test('Bruker uten oppf.perioder og ikke under oppf skjuler store deler av appen', () => {
        useFetchOppfolging.data.underOppfolging = false;
        useFetchOppfolging.data.oppfolgingsPerioder = [];
        jest.spyOn(AppContext, 'useOppfolgingContext').mockImplementation(() => useFetchOppfolging);
        const wrapper = mount(
            <MemoryRouter>
                <DialogListe />
            </MemoryRouter>
        );
        expect(wrapper.find(NyDialogLink).exists()).toBeFalsy();
        expect(wrapper.find(Dialog).exists()).toBeFalsy();
        expect(wrapper.find(DialogPreview).exists()).toBeFalsy();
    });
    test('Bruker ikke under oppf. skjuler knapper/checkbox', () => {
        useFetchOppfolging.data.underOppfolging = false;
        useFetchOppfolging.data.oppfolgingsPerioder = [
            {
                aktorId: '1234567988888',
                veileder: false,
                startDato: '2017-01-30T10:46:10.971+01:00',
                sluttDato: '2017-12-31T10:46:10.971+01:00',
                begrunnelse: null
            }
        ];
        jest.spyOn(AppContext, 'useOppfolgingContext').mockImplementation(() => useFetchOppfolging);
        jest.spyOn(DialogProvider, 'useDialogContext').mockImplementation(() => useDialogContext);
        const wrapper = mount(
            <MemoryRouter>
                <DialogListe />
            </MemoryRouter>
        );
        expect(wrapper.find(NyDialogLink).exists()).toBeFalsy();
        expect(wrapper.find(DialogPreview).exists()).toBeTruthy();
    });
    test('Bruker under oppf, elementer synes', () => {
        useFetchOppfolging.data.underOppfolging = true;
        useFetchOppfolging.data.oppfolgingsPerioder = [
            {
                aktorId: '1234567988888',
                veileder: false,
                startDato: '2017-01-30T10:46:10.971+01:00',
                sluttDato: '2017-12-31T10:46:10.971+01:00',
                begrunnelse: null
            }
        ];
        jest.spyOn(AppContext, 'useOppfolgingContext').mockImplementation(() => useFetchOppfolging);
        jest.spyOn(AppContext, 'useHarNivaa4Context').mockImplementation(() => ({
            harNivaa4: true,
            hasError: false,
            isPending: false
        }));
        jest.spyOn(DialogProvider, 'useDialogContext').mockImplementation(() => useDialogContext);
        const wrapper = mount(
            <MemoryRouter>
                <DialogOversikt />
            </MemoryRouter>
        );
        expect(wrapper.find(NyDialogLink).exists()).toBeTruthy();
        expect(wrapper.find(DialogPreview).exists()).toBeTruthy();
    });
});

describe('<Dialog/>', () => {
    test('Bruker ikke under oppf. skjuler dialogcontroller og viser fortsatt henvendelser', () => {
        useFetchOppfolging.data.underOppfolging = false;
        useFetchOppfolging.data.oppfolgingsPerioder = [
            {
                aktorId: '1234567988888',
                veileder: false,
                startDato: '2017-01-30T10:46:10.971+01:00',
                sluttDato: '2017-12-31T10:46:10.971+01:00',
                begrunnelse: null
            }
        ];
        jest.spyOn(DialogProvider, 'useDialogContext').mockImplementation(() => useDialogContext);
        jest.spyOn(AppContext, 'useOppfolgingContext').mockImplementation(() => useFetchOppfolging);
        Element.prototype.scrollIntoView = () => {};
        const wrapper = mount(
            <MemoryRouter initialEntries={['/1']}>
                <DialogContainer />
            </MemoryRouter>
        );
        expect(wrapper.find(DialogInputBox).exists()).toBeTruthy();
        expect(wrapper.find(HenvendelseList).exists()).toBeTruthy();
        expect(wrapper.find(Checkbox).exists()).toBeFalsy();
    });
    test('Bruker under oppf. viser komponenter i Dialog', () => {
        useFetchOppfolging.data.underOppfolging = true;
        useFetchOppfolging.data.oppfolgingsPerioder = [
            {
                aktorId: '1234567988888',
                veileder: false,
                startDato: '2017-01-30T10:46:10.971+01:00',
                sluttDato: '2017-12-31T10:46:10.971+01:00',
                begrunnelse: null
            }
        ];
        jest.spyOn(AppContext, 'useOppfolgingContext').mockImplementation(() => useFetchOppfolging);
        jest.spyOn(AppContext, 'useUserInfoContext').mockImplementation(() => userInfo);
        Element.prototype.scrollIntoView = () => {};
        const wrapper = mount(
            <MemoryRouter initialEntries={['/1']}>
                <DialogContainer />
            </MemoryRouter>
        );
        expect(wrapper.find(DialogHeader).exists()).toBeTruthy();
        expect(wrapper.find(DialogInputBox).exists()).toBeTruthy();
        expect(wrapper.find(HenvendelseList).exists()).toBeTruthy();
    });
});
