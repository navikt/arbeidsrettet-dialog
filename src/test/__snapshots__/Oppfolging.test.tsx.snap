// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<DialogContainer/> > Bruker under oppf, elementer synes 1`] = `
<body>
  <div>
    <div
      class="border-r border-border-divider _dialogOversikt_d95bf1"
    >
      <div
        class="_header_d95bf1"
      >
        <a
          class="navds-link _dintnav_d95bf1"
          href="http://www.intern.dev.nav.no/minside"
        >
          <svg
            fill="none"
            focusable="false"
            height="1em"
            role="img"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clip-rule="evenodd"
              d="M14.53 5.97a.75.75 0 0 1 0 1.06L9.56 12l4.97 4.97a.75.75 0 1 1-1.06 1.06l-5.5-5.5a.75.75 0 0 1 0-1.06l5.5-5.5a.75.75 0 0 1 1.06 0Z"
              fill="currentColor"
              fill-rule="evenodd"
            />
          </svg>
          Min side
        </a>
        <h1
          class="_tittel_d95bf1 navds-heading navds-heading--undefined"
        >
          Dialog med veilederen din
        </h1>
        <div
          class="flex mt-4 justify-between mr-6 items-center"
        >
          <div
            class="_header_1c099b"
          >
            <button
              class="navds-button navds-button--secondary navds-button--medium"
            >
              <span
                class="navds-button__icon"
              >
                <svg
                  fill="none"
                  focusable="false"
                  height="1em"
                  role="img"
                  viewBox="0 0 24 24"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    clip-rule="evenodd"
                    d="M12.75 5.5a.75.75 0 0 0-1.5 0v5.75H5.5a.75.75 0 0 0 0 1.5h5.75v5.75a.75.75 0 0 0 1.5 0v-5.75h5.75a.75.75 0 0 0 0-1.5h-5.75V5.5Z"
                    fill="currentColor"
                    fill-rule="evenodd"
                  />
                </svg>
              </span>
              <span
                aria-live="polite"
                class="navds-label"
              >
                Ny dialog
              </span>
            </button>
          </div>
          <a
            href="/informasjon"
          >
            <span
              class="flex"
            >
              <span>
                Om dialog
              </span>
              <svg
                class="w-6 h-6"
                fill="none"
                focusable="false"
                height="1em"
                role="img"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  clip-rule="evenodd"
                  d="M10.25 9.5c0-.58.19-1.004.468-1.282S11.42 7.75 12 7.75c.58 0 1.004.19 1.282.468s.468.702.468 1.282c0 .158-.064.35-.25.61-.192.269-.457.537-.78.86l-.022.02c-.297.298-.647.648-.918 1.027-.282.395-.53.89-.53 1.483v.5a.75.75 0 0 0 1.5 0v-.5c0-.158.064-.35.25-.61.192-.269.457-.537.78-.86l.022-.02c.297-.298.647-.648.918-1.027.282-.395.53-.89.53-1.483 0-.92-.31-1.746-.907-2.343S12.92 6.25 12 6.25c-.92 0-1.746.31-2.343.907S8.75 8.58 8.75 9.5a.75.75 0 0 0 1.5 0Zm1.75 6a1 1 0 1 0 0 2 1 1 0 0 0 0-2Z"
                  fill="currentColor"
                  fill-rule="evenodd"
                />
              </svg>
            </span>
          </a>
        </div>
      </div>
      <div
        aria-label="Dialoger"
        class="_dialogListe_d95bf1"
        role="navigation"
      >
        <div>
          <div
            aria-label="Dialogliste"
            role="list"
          >
            <div
              data-flip-config="{\\"translate\\":true,\\"scale\\":true,\\"opacity\\":true}"
              data-flip-id="1"
              data-portal-key="portal"
              role="listitem"
            >
              <a
                class="navds-panel navds-link-panel my-2 border _preview_f5da6d navds-panel--border"
                href="/1"
              >
                <div
                  class="navds-link-panel__content"
                >
                  <div
                    class="flex"
                  >
                    <div
                      class="_markoerLest_f5da6d"
                    />
                    <div
                      aria-hidden="true"
                      class="_ikon_ba0748"
                    >
                      <svg
                        class="w-10 h-10"
                        focusable="false"
                        style="enable-background: new 0 0 24 24;"
                        viewBox="0 0 24 24"
                        x="0px"
                        xml:space="preserve"
                        xmlns="http://www.w3.org/2000/svg"
                        y="0px"
                      >
                        <title>
                          Dialog om aktivitet
                        </title>
                        <style
                          type="text/css"
                        >
                          
	.st0{fill:none;}
	.st1{fill:#3E3832;}

                        </style>
                        <g>
                          <path
                            class="st0"
                            d="M18.1,4H3.9C2.3,4,1,5.3,1,6.9v9.2C1,17.7,2.3,19,3.9,19H11v-7h10V6.9C21,5.3,19.7,4,18.1,4z"
                          />
                          <path
                            class="st1"
                            d="M3.9,19C2.3,19,1,17.7,1,16.1V6.9C1,5.3,2.3,4,3.9,4h14.2C19.7,4,21,5.3,21,6.9V12h1V6.9C22,4.7,20.3,3,18.1,3 H3.9C1.7,3,0,4.7,0,6.9v9.2C0,18.3,1.7,20,3.9,20H11v-1H3.9z"
                          />
                        </g>
                        <g>
                          <path
                            class="st1"
                            d="M18.5,12.8c-3.2,0-5.8,2.1-5.8,4.7c0,1.2,0.5,2.3,1.5,3.1l-1,2c-0.1,0.2-0.1,0.4,0.1,0.5 c0.1,0.1,0.2,0.1,0.3,0.1c0.1,0,0.2,0,0.2,0l2.9-1.3c0.6,0.1,1.2,0.2,1.8,0.2c3.2,0,5.8-2.1,5.8-4.7C24.2,14.9,21.7,12.8,18.5,12.8 z M18.5,21.2c-0.6,0-1.2-0.1-1.7-0.2c-0.1,0-0.2,0-0.3,0l-1.8,0.8l0.6-1.1c0.1-0.2,0-0.5-0.1-0.6c-0.9-0.7-1.4-1.6-1.4-2.6 c0-2.1,2.1-3.7,4.8-3.7s4.8,1.7,4.8,3.7C23.3,19.5,21.1,21.2,18.5,21.2z"
                          />
                        </g>
                        <g>
                          <path
                            class="st1"
                            d="M3.6,8h9.8C13.7,8,14,8.3,14,8.6v0c0,0.3-0.3,0.6-0.6,0.6H3.6C3.3,9.2,3,8.9,3,8.6v0C3,8.3,3.3,8,3.6,8z"
                          />
                        </g>
                        <g>
                          <path
                            class="st1"
                            d="M3.6,11h9.8c0.3,0,0.6,0.3,0.6,0.6v0c0,0.3-0.3,0.6-0.6,0.6H3.6c-0.3,0-0.6-0.3-0.6-0.6v0 C3,11.3,3.3,11,3.6,11z"
                          />
                        </g>
                        <g>
                          <path
                            class="st1"
                            d="M3.6,14h1.8C5.7,14,6,14.3,6,14.6v0c0,0.3-0.3,0.6-0.6,0.6H3.6c-0.3,0-0.6-0.3-0.6-0.6v0C3,14.3,3.3,14,3.6,14 z"
                          />
                        </g>
                      </svg>
                    </div>
                    <div
                      class="flex-1 "
                    >
                      <p
                        class="hidden navds-body-short"
                      >
                        Dialog om aktivitet
                      </p>
                      <p
                        class="_heading_f5da6d navds-heading navds-heading--undefined"
                      >
                        Memes
                      </p>
                      <p
                        class="_dato_f5da6d navds-body-short"
                      >
                        28.01.2018
                      </p>
                      <p
                        class="hidden navds-body-short"
                      >
                        2 meldinger i dialogen
                      </p>
                    </div>
                    <p
                      aria-hidden="true"
                      class="navds-body-short"
                    >
                      2
                    </p>
                    <div />
                  </div>
                </div>
                <svg
                  aria-hidden="true"
                  class="navds-link-panel__chevron"
                  fill="none"
                  focusable="false"
                  height="1em"
                  role="img"
                  viewBox="0 0 24 24"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    clip-rule="evenodd"
                    d="m17.414 12-7.707 7.707-1.414-1.414L14.586 12 8.293 5.707l1.414-1.414L17.414 12Z"
                    fill="currentColor"
                    fill-rule="evenodd"
                  />
                </svg>
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;
